### PR TITLE
[Backend] - Replace sidekiq with SQS Queues

### PR DESCRIPTION
**Pull request by sunil-sadasivan**
This replaces sidekiq for SQS queues.  #3182 

Opening this now, but will want to work closely with Artem to test and deploy this carefully in prod. 

**Important**  in order to deploy this, we'll need to deploy the shoryuken workers

```
#UAT env
shoryuken start -q uat_high_priority uat_low_priority -R

#Prod env
shoryuken start -q prod_high_priority prod_low_priority -R
```
(I've created an issue in appeals-deployment here https://github.com/department-of-veterans-affairs/appeals-deployment/issues/869)

To test locally:
```
$ rails c
> HeartbeatTasksJob.perform_later

# then run:
shoryuken start -q development_high_priority development_low_priority -R
```

### CODE REVIEW

**Comment by evankroske** [comment]
We should create the SQS queues using Terraform. I can help

**Comment by sunil-sadasivan** [comment]
@evankroske yes!  Will def need to do that.  After talking with JD, I've realized I'll first want to replace all sidekiq-cron jobs with CW/Lambda and then move to replace redis with SQS queues.

File: app/jobs/start_certification_job.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 class StartCertificationJob < ActiveJob::Base
+  # TODO(sunil): for now use sidekiq. when sqs works well
+  # use shoryuken instead of sidekiq
   queue_as :default
+  self.queue_adapter = :sidekiq
```
Comment (joofsh) [inline app/jobs/start_certification_job.rb]: nice 👍

File: config/application.rb [hunk 2]
```diff
@@ -36,7 +36,8 @@ class Application < Rails::Application
 
     config.google_analytics_account = nil
 
-    config.active_job.queue_adapter = :sidekiq
+    config.active_job.queue_adapter = :shoryuken
+    config.active_job.queue_name_prefix = ENV['DEPLOY_ENV'] || Rails.env
```
Comment (joofsh) [inline config/application.rb]: Don't we have a helper method that abstracts this?

File: config/environments/demo.rb [hunk 3]
```diff
@@ -84,4 +84,6 @@
 
   ENV["SIDEKIQ_USERNAME"] ||= "caseflow"
   ENV["SIDEKIQ_PASSWORD"] ||= "caseflow"
+
+  ENV["AWS_REGION"] ||= "us-gov-west-1"
```
Comment (joofsh) [inline config/environments/demo.rb]: If this is the same in every environment, we can move this up to the `application.rb`

File: config/application.rb (hunk 2 above)
Comment (sunil-sadasivan) [reply config/application.rb]: @joofsh, unfortunately not (that I'm aware of). tbh, was surprised about this, though, I can write up one.

File: config/environments/demo.rb (hunk 3 above)
Comment (sunil-sadasivan) [reply config/environments/demo.rb]: 👍

**Comment by sunil-sadasivan** [comment]
Broke this PR into 3 different parts. 

https://github.com/department-of-veterans-affairs/caseflow/pull/3354
https://github.com/department-of-veterans-affairs/caseflow/pull/3385

**Pull request closed without merging**